Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -1059,8 +1060,94 @@ void PcapLiveDevice::setDefaultGateway()
 			PCPP_LOG_ERROR("Error retrieving default gateway address: " << e.what());
 		}
 	}
-#elif defined(__APPLE__) || defined(__FreeBSD__)
-	std::string command = "netstat -nr | grep default | grep " + m_Name;
+#elif defined(__APPLE__)
+
+	//route message struct for communication in APPLE device
+	struct BSDRoutingMessage{
+		struct	rt_msghdr header;
+		char	messageSpace[512];
+	};
+
+	struct BSDRoutingMessage routingMessage;
+	char* spacePtr = routingMessage.messageSpace;
+	// It creates a raw socket that can be used for routing-related operations
+	int sockfd = socket(PF_ROUTE, SOCK_RAW, 0);
+	if (sockfd < 0) {
+		PCPP_LOG_ERROR("Error retrieving default gateway address: couldn't get open routing socket");
+		return ;
+	}
+
+    memset(reinterpret_cast<char*>(&routingMessage), 0, sizeof(routingMessage));
```

Comment:
nit: File uses tabs instead of spaces for indenting.
```suggestion
	memset(reinterpret_cast<char*>(&routingMessage), 0, sizeof(routingMessage));
```